packages/ra-core/src/inference/assertions.ts

```diff
@@ -4,27 +4,41 @@ export const isNumeric = (value: any) =>
     !isNaN(parseFloat(value)) && isFinite(value);
 export const valuesAreNumeric = (values: any[]) => values.every(isNumeric);
 
-export const isInteger = (value: any) => Number.isInteger(value);
+export const isInteger = (value: any) =>
+    Number.isInteger(value) || !isNaN(parseInt(value));
 export const valuesAreInteger = (values: any[]) => values.every(isInteger);
 
-export const isBoolean = (value: any) => typeof value === 'boolean';
+export const isBoolean = (value: any) =>
+    typeof value === 'boolean' ||
+    ['true', 'false'].includes(value.toLowerCase());
 export const valuesAreBoolean = (values: any[]) => values.every(isBoolean);
 
 export const isString = (value: any) => typeof value === 'string';
 export const valuesAreString = (values: any[]) => values.every(isString);
 
 const HtmlRegexp = /<([A-Z][A-Z0-9]*)\b[^>]*>(.*?)<\/\1>/i;
-export const isHtml = (value: any) => HtmlRegexp.test(value);
+export const isHtml = (value: any) => !value || HtmlRegexp.test(value);
 export const valuesAreHtml = (values: any[]) => values.every(isHtml);
 
+const UrlRegexp = /http(s*):\/\/.*/i;
+export const isUrl = (value: any) => !value || UrlRegexp.test(value);
+export const valuesAreUrl = (values: any[]) => values.every(isUrl);
+
+const ImageUrlRegexp = /.*\.(jpeg|jpg|jfif|pjpeg|pjp|png|svg|gif|webp|apng|bmp|ico|cur|tif|tiff)/i;
+export const isImageUrl = (value: any) => !value || ImageUrlRegexp.test(value);
+export const valuesAreImageUrl = (values: any[]) => values.every(isImageUrl);
+
+export const isEmail = (value: any) => !value || value.indexOf('@') > -1;
```

Comment:
very weak: 'foo@bar@com' is an email?

Reply:
The only thing you can check is the presence of the `@`. I may add a count but that will be all